networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/forks/versions/GossipForkSubscriptionsEip4844.java

```diff
@@ -0,0 +1,104 @@
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.networking.eth2.gossip.BlockAndBlobsSidecarGossipManager;
+import tech.pegasys.teku.networking.eth2.gossip.encoding.GossipEncoding;
+import tech.pegasys.teku.networking.eth2.gossip.topics.OperationProcessor;
+import tech.pegasys.teku.networking.p2p.discovery.DiscoveryNetwork;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.datastructures.attestation.ValidateableAttestation;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.eip4844.SignedBeaconBlockAndBlobsSidecar;
+import tech.pegasys.teku.spec.datastructures.operations.AttesterSlashing;
+import tech.pegasys.teku.spec.datastructures.operations.ProposerSlashing;
+import tech.pegasys.teku.spec.datastructures.operations.SignedBlsToExecutionChange;
+import tech.pegasys.teku.spec.datastructures.operations.SignedVoluntaryExit;
+import tech.pegasys.teku.spec.datastructures.operations.versions.altair.SignedContributionAndProof;
+import tech.pegasys.teku.spec.datastructures.operations.versions.altair.ValidateableSyncCommitteeMessage;
+import tech.pegasys.teku.spec.datastructures.state.Fork;
+import tech.pegasys.teku.spec.datastructures.state.ForkInfo;
+import tech.pegasys.teku.storage.client.RecentChainData;
+
+public class GossipForkSubscriptionsEip4844 extends GossipForkSubscriptionsCapella {
+
+  private final OperationProcessor<SignedBeaconBlockAndBlobsSidecar> gossipedBlockAndBlobsProcessor;
+
+  private BlockAndBlobsSidecarGossipManager blockAndBlobsSidecarGossipManager;
+
+  public GossipForkSubscriptionsEip4844(
+      final Fork fork,
+      final Spec spec,
+      final AsyncRunner asyncRunner,
+      final MetricsSystem metricsSystem,
+      final DiscoveryNetwork<?> discoveryNetwork,
+      final RecentChainData recentChainData,
+      final GossipEncoding gossipEncoding,
+      final OperationProcessor<SignedBeaconBlockAndBlobsSidecar> gossipedBlockAndBlobsProcessor,
+      final OperationProcessor<ValidateableAttestation> attestationProcessor,
+      final OperationProcessor<ValidateableAttestation> aggregateProcessor,
+      final OperationProcessor<AttesterSlashing> attesterSlashingProcessor,
+      final OperationProcessor<ProposerSlashing> proposerSlashingProcessor,
+      final OperationProcessor<SignedVoluntaryExit> voluntaryExitProcessor,
+      final OperationProcessor<SignedContributionAndProof>
+          signedContributionAndProofOperationProcessor,
+      final OperationProcessor<ValidateableSyncCommitteeMessage>
+          syncCommitteeMessageOperationProcessor,
+      final OperationProcessor<SignedBlsToExecutionChange>
+          signedBlsToExecutionChangeOperationProcessor) {
+    super(
+        fork,
+        spec,
+        asyncRunner,
+        metricsSystem,
+        discoveryNetwork,
+        recentChainData,
+        gossipEncoding,
+        Optional.empty(),
```

Comment:
If I got it correctly, we should continue gossiping for 2 epochs after transition.
See: https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/p2p-interface.md#transitioning-the-gossip
Discussion https://github.com/ethereum/consensus-specs/pull/3090#discussion_r1020463652

Reply:
Ahh yes that's a good point.. I missed that issue

Reply:
I'm going to change the way I do this. Thinking about introducing a more structural unsubscription pattern (see comment below to luca's feedback)

Reply:
as we saw, everything is nicely handled by `GossipForkManager::configureGossipForEpoch`